source/data-handling-database-R.Rmd

```diff
@@ -0,0 +1,150 @@
+
+```{r queryexample}
+subset_meting <- sqlQuery(my_connection, 
+"SELECT meet.Cor_X
+	 , meet.COr_Y
+	 , meet.MetingStatusCode
+	 , tax.NaamNederlands
+	 , tax.NaamWetenschappelijk
+	 , waar.IFBLHokID
+FROM  [dbo].[tblMeting] meet
+	LEFT JOIN [dbo].tblTaxon tax ON tax.ID = meet.TaxonID
+	LEFT JOIN [dbo].tblWaarneming waar ON waar.ID = meet.WaarnemingID
+WHERE meet.Cor_X IS NOT NULL
+	AND meet.Cor_X != 0
+	AND tax.NaamNederlands LIKE 'Wilde hyacint'")
+head(subset_meting)
+```
+
+If we need this query regularly, but each time using a different `tax.NaamNederlands` (only the name changes), it is worthwhile to invest some time in the creation of a small custom function that uses this query as a template. Let's create a function `flora_records_on_dutch_name` that takes a valid database connection and a given dutch name and returns the relevant subset of the data for this query:
+
+```{r querytemplate}
+
+flora_records_on_dutch_name <- function(dbase_connection, dutch_name) {
+     sqlQuery(dbase_connection, sprintf(
+            "SELECT meet.Cor_X
+            	 , meet.COr_Y
+            	 , meet.MetingStatusCode
+            	 , tax.NaamNederlands
+            	 , tax.NaamWetenschappelijk
+            	 , waar.IFBLHokID
+            FROM  [dbo].[tblMeting] meet
+            	LEFT JOIN [dbo].tblTaxon tax ON tax.ID = meet.TaxonID
+            	LEFT JOIN [dbo].tblWaarneming waar ON waar.ID = meet.WaarnemingID
+            WHERE meet.Cor_X IS NOT NULL
+            	AND meet.Cor_X != 0
+            	AND tax.NaamNederlands LIKE '%s'", dutch_name))
+}
+
+```
+
+Hence, instead of copy-pasting the whole query each time (which could be error-prone), we can reuse the function for different names:
+
+```{r}
+hyacint <- flora_records_on_dutch_name(my_connection, "Wilde hyacint")
+head(hyacint)
+```
+
+```{r}
+bosanemoon <- flora_records_on_dutch_name(my_connection, "Bosanemoon")
+head(bosanemoon)
+```
+
+**Remark:** Do not forget to close your connection when done finished. 
+
+```{r closeit}
+close(my_connection)
+```
+
+## The `sprintf` function {#sprintf}
+
+In order to accomplish the re-usage of a query for different input names (`dutch_name`), the `sprintf` function is used. The `sprintf` function provides the ability to combine text and variable values in a single charactor string (i.e. the query to execute). For each variable name required in the query (any part of your query you want to have interchangeable), a representation in the query is given by a `%` in combination with the data type you want to represent. The latter is an agreed format conversion, for example `%s` is a character string (e.g. the example of `dutch_name`) and `%d` is an integer value (e.g. `2`). More information about the different symbols is given in the documentation of `sprintf` (type `?sprintf` in the console). 
+
+```{r sprintfex1}
+aname <- 'Jan'
```

Comment:
separate the words with underscores to make it more readable: `a_name` vs `aname`